crates/core/tedge/src/main.rs

```diff
@@ -139,6 +148,21 @@ where
         }
     }
 
+    if matches!(
+        executable_name.as_deref(),
+        Some("file" | "tedge-file-log-plugin")
+    ) {
+        // the file log plugin must be treated apart
+        // as we want to exit 1 and not 2 when the command line cannot be parsed
+        match FileLogCli::try_parse() {
+            Ok(cli) => return TEdgeOptMulticall::Component(Component::TedgeFileLogPlugin(cli)),
+            Err(e) => {
+                eprintln!("{}", RichFormatter::format_error(&e));
```

Comment:
This `RichFormatter` completely messes up the error!

```
$ ls -lh file
lrwxrwxrwx 1 didier didier   55 Sep 26 09:30 file -> target/debug/tedge

$ ./file       
error: unknown cause
For more information, try '--help'.

$ ./file --help
error: unknown cause
For more information, try '--help'.
```

The fix is simple:


```suggestion
                eprintln!("{e}");
```

=> I pushed a fixup commit

Reply:
I pushed a fixup: https://github.com/thin-edge/thin-edge.io/pull/3762/commits/fbdb363ef96af61df019ac32d9daa6ebd9afd3b7